index.js

```diff
@@ -1,33 +1,212 @@
+      request(url, function(error, response, body) {
+        if(!error && response.statusCode == 200){
+          resolve(JSON.parse(body).data.schema);
+        }else{
+          console.log("An error occurred - " + response.statusCode);
+          reject("Error occurred while getting schema - " + response.statusCode);
+        }
+      })
+    })
+}
 
+//use avro to deserialize
+var avro_decoded_data = function(schema_data, record){
+  const type = avro.parse(schema_data);
+  var decoded = new Buffer(record, 'base64');
+  var verify = type.fromBuffer(decoded);
+  return JSON.parse(verify);
+}
 
+//get full bib/item details for each bib/item id
+var resource_in_detail = function(id, isBib){
+  return new Promise(function(resolve, reject){
+    var operation = retry.operation({
+        retries: 5,
+        factor: 3,
+        minTimeout: 1 * 1000,
+        maxTimeout: 60 * 1000,
+        randomize: true,
+    });
+    operation.attempt(function(currentAttempt) {
+      if(isBib){
+        console.log('Requesting for bib info');
+        wrapper.requestSingleBib(id, (errorBibReq, results) => {
+        getResult(errorBibReq, results, true, id, operation, currentAttempt)
+            .then(function(entry){
+              resolve (entry);
+            })
+            .catch (function(e) {
+              reject (e);
+            });
+        });
+      } else{
+        console.log('Requesting for item info');
+        var itemIds = [id];
+        wrapper.requestMultiItemBasic(itemIds, (errorItemReq, results) => {
+          getResult(errorItemReq, results, false, itemIds, operation, currentAttempt)
+            .then(function(entry){
+              resolve (entry);
+            })
+            .catch (function(e) {
+              reject (e);
+            });
+        });
+      }
+    });
+  });
+}
 
-// var params = {
-//   StreamName: 'SierraBibPostRequestTest' /* required */
-// };
-// kinesis.describeStream(params, function(err, data) {
-//   if (err) console.log(err, err.stack); // an error occurred
-//   else     console.log(JSON.stringify(data,null,2));           // successful response
+//get bib or item based on prama passed
+var getResult = function(errorResourceReq, results, isBib, resourceId, operation, attemptNumber){
+  return new Promise(function(resolve, reject){
+    if(errorResourceReq){
+        if(errorResourceReq.httpStatus !== null && errorResourceReq.httpStatus === 401){
+          console.log("This is a token issue. Going to renew token");
+          if(isBib) {
+            console.log('Number of attempts made for bib ' + resourceId + ' - ' + attemptNumber);
+          } else{
+            console.log('Number of attempts made for item ' + resourceId + ' - ' + attemptNumber);
+          }
+          set_wrapper_access_token()
+            .then(function (access_token){
+              wrapper_access_token = access_token;
+              if(operation.retry(errorResourceReq)){
+                return;
```

Comment:
If you do `return;` here, then the code between lines 166-171 will never be executed.